ownCloud/Settings/Passcode/PasscodeViewController.swift

```diff
@@ -199,36 +199,44 @@ class PasscodeViewController: UIViewController, Themeable {
 
 	// MARK: - Actions
 	@IBAction func appendDigit(_ sender: UIButton) {
-		if !keypadButtonsEnabled || keypadButtonsHidden {
-			return
-		}
-
-		if let currentPasscode = passcode {
-			// Enforce length limit
-			if currentPasscode.count < passcodeLength {
-				self.passcode = currentPasscode + String(sender.tag)
-			}
-		} else {
-			self.passcode = String(sender.tag)
-		}
-
-		// Check if passcode is complete
-		if let passcode = passcode {
-			if passcode.count == passcodeLength {
-				// Delay to give feedback to user after the last digit was added
-				OnMainThread(after: 0.1) {
-					self.completionHandler?(self, passcode)
-				}
-			}
-		}
+        appendDigit(digit: String(sender.tag))
 	}
+    
+    private func appendDigit(digit: String) {
+        if !keypadButtonsEnabled || keypadButtonsHidden {
+            return
+        }
+        
+        if let currentPasscode = passcode {
+            // Enforce length limit
+            if currentPasscode.count < passcodeLength {
+                self.passcode = currentPasscode + String(digit)
+            }
+        } else {
+            self.passcode = String(digit)
+        }
+        
+        // Check if passcode is complete
+        if let passcode = passcode {
+            if passcode.count == passcodeLength {
+                // Delay to give feedback to user after the last digit was added
+                OnMainThread(after: 0.1) {
+                    self.completionHandler?(self, passcode)
+                }
+            }
+        }
+    }
 
 	@IBAction func deleteLastDigit(_ sender: UIButton) {
-		if passcode != nil, passcode!.count > 0 {
-			passcode?.removeLast()
-			updatePasscodeDots()
-		}
+        deleteLastDigit()
 	}
+    
+    private func deleteLastDigit() {
+        if passcode != nil, passcode!.count > 0 {
```

Comment:
@hosy wouldn't `if passcode?.count > 0` be shorter?

Reply:
Does not work, because of a compiler error

Reply:
@hosy right.. i was wrong